crypto/fipsmodule/bn/montgomery.c

```diff
@@ -418,6 +450,82 @@ static int bn_mod_mul_montgomery_fallback(BIGNUM *r, const BIGNUM *a,
   return ret;
 }
 
+
+#if defined(OPENSSL_BN_ASM_MONT)
+
+// Perform montgomery multiplication using s2n-bignum functions. The arguments
+// are equivalent to the arguments of bn_mul_mont.
+// montgomery_s2n_bignum_mul_mont works only if num is a multiple of 8. For
+// num = 32 or num = 16, this uses faster primitives in s2n-bignum.
+// Additionally, montgomery_s2n_bignum_mul_mont allocates arrays at a stack, and
+// large num leads to out of bounds accesses of the arrays.
+// montgomery_use_s2n_bignum(num) must be called in advance to check these
+// conditions.
+static void montgomery_s2n_bignum_mul_mont(BN_ULONG *rp, const BN_ULONG *ap,
+                                           const BN_ULONG *bp,
+                                           const BN_ULONG *np,
+                                           const BN_ULONG *n0, size_t num) {
+
+#if defined(BN_MONTGOMERY_USE_S2N_BIGNUM)
```

Comment:
This `#if` can replace the one above, 'OPENSSL_BN_ASM_MONT', so there is only one `#if..#else` pair, since it's more constrained.